### DISCUSSION TITLE
New bladebit 2.0 vs madmax+ramcache

### DISCUSSION THREAD

**Discussion by rnizametdinov** [Chia-Network/bladebit#186] [Show and tell]
I have 5900x ryzen with 64gb ram and made some test.
I monitor  writen gb of data on ssd drive through S.M.A.R.T /

1) madmax+ramcache 55gb , one ssd divided on 
 -2 = 128gb partition with 55 gb cache 
 -t =  big volume on same drive
 Plot time 42 min and 816Gb write

1) bladebit alpha --cache 55gb  
 Plot time 41 min and 831Gb write

So we have same write amount and same plot time. Not bad for alpha. 
bladebit minus - it peak consamption 440Gb on ssd drive vs 260Gb by madmax, but it not critical in most cases.

**Comment by rnizametdinov** [comment]
1. madmax+ramcache 120gb , one ssd divided on
-2 = 128gb partition with 120 gb cache
-t = big volume on same drive
Plot time 35 min and 387Gb write !!!!

2. bladebit alpha --cache 120gb
Plot time 37 min and 520Gb write :(((((((

MAdmax win on RAM cache 120Gb : 387Gb < 520Gb write to drive. 
Writen umount monitored by SMART

**Comment by ruixingchen** [comment]
I still use madmax because 128G RAM is much more easier than 512G ram，I use 5950X and 7980X, with 128G RAM they all work well

**Reply by rnizametdinov** [reply]
>I still use madmax because 128G RAM is much more easier than 512G ram

New alfa version of bladebit dont need 512 ram. It can use any amount of ram for cache (i tested with 64gb of ram). More ram you have - less write to disk on plotting.

**Reply by ruixingchen** [reply]
> 

do you mean BBV2 can use RAM dynamicly? like if I have 256 ram, it will reduce disk writes, and if I have 416G ram it will use ram only?

**Reply by rnizametdinov** [reply]
Yes, look here
https://github.com/Chia-Network/bladebit/wiki/Bladebit-2-User-Guide-(WIP)
"you can add DRAM cache to reduce SSD writes with --cache <amount> <units>
an example below
./bladebit -t <thread count> -f <farmer key> -c <contract address> diskplot
--cache 32G -t1 /mnt/ssd /mnt/hdd   "

Write --cache 246G if you have 256g ram.

But... if you have 256G RAM you can use madmax almost in memory.
You must activate SWAP, make tmpfs for 256Gb in ram, and set only one temp
dir with this ramdrive on madmax.

**Reply by ruixingchen** [reply]
> Yes, look here https://github.com/Chia-Network/bladebit/wiki/Bladebit-2-User-Guide-(WIP) "you can add DRAM cache to reduce SSD writes with --cache an example below ./bladebit -t -f -c diskplot --cache 32G -t1 /mnt/ssd /mnt/hdd "
> 
> Write --cache 246G if you have 256g ram.
> 
> But... if you have 256G RAM you can use madmax almost in memory. You must activate SWAP, make tmpfs for 256Gb in ram, and set only one temp dir with this ramdrive on madmax.

thank you for notice me that, currently I use 128G and a nvme disk, works well, when BBV2 comes to beta, I will have a try, I do not want to use 512G server, it is hard to re-sell

**Comment by turbo944s2** [comment]
I just gave it a shot and got this error 

terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted (core dumped)

**Reply by turbo944s2** [reply]
Found the problem. I didnt put in a destination drive after the SSD. Best time was 17.7 minutes on a TR 3970x 2666 mhz. Plot would not finish with my overclock settings at 3200. I supposed it is due to the memory being half samsung and half hynix.

**Comment by rnizametdinov** [comment]
Alpha 2 bladebit

madmax+ram 110GB ram-disk
Plot time 35 min and 387Gb write on ssd
Peak ssd space consumption 256GB
 
bladebit alpha --cache 100GB
Plot time 33 min and 434 GB write, much better than alpha 1,

MAdmax win on ssd write : 387GB < 434GB, but bladebit faster.
And i think bladebit speed can be tuned.
Ps/ Written umount monitored by intel s3610 SMART

**Reply by ruixingchen** [reply]
I use bladebit V1 and now it works well, I will sell all my SSDs and use RAM only

**Reply by ruixingchen** [reply]
I tested BBV2 alpha 2 on my 7960X+128G plotter, surprisely BB is slower than madmax, I use the sample 99G ram cache command, centos 8 stream

**Comment by FuzeGuy** [comment]
>>>No contest<<<   
 [Bladebit with MMX v2.0.0-mmx) BB wins by a landslide. 

Threadripper Pro 3955wx 128GB 3200MHz memory
16 cores/32 (31 used) threads, 120G cache, Sam 980 Pro 2G t1 [>]([url](url)) K32 MMX Plot

Total plot I/O wait time: 40.98 seconds.
Finished plotting in 886.46 seconds ( 14.8 minutes )

Best prior MM plotting 21.5 min (typical) w/2TB t1 & 1TB t2 Sam 980 Pros
>>>>>> Bladebit Disk v2 >>> 32% improved plot time vs MM (Chia k32)

Plot Log >> [best 14.8 min k32 BB Disk V2 plot log.txt](https://github.com/Chia-Network/bladebit/files/9656374/best.14.8.min.k32.BB.Disk.V2.plot.log.txt)
